internal/proxy/search_util.go

```diff
@@ -163,9 +189,37 @@ func parseSearchInfo(searchParamsPair []*commonpb.KeyValuePair, schema *schemapb
 	}
 
 	// 4. parse search param str
-	searchParamStr, err := funcutil.GetAttrByKeyFromRepeatedKV(SearchParamsKey, searchParamsPair)
+	searchParamStr, err := funcutil.GetAttrByKeyFromRepeatedKV(ParamsKey, searchParamsPair)
+	var searchParamMap map[string]any
 	if err != nil {
 		searchParamStr = ""
+	} else {
+		err = json.Unmarshal([]byte(searchParamStr), &searchParamMap)
+		if err != nil {
+			return &SearchInfo{planInfo: nil, offset: 0, isIterator: false, parseError: err}
+		}
+	}
+
+	// related with https://github.com/milvus-io/milvus/issues/37972
+	// before 2.5.1, all key in ParamsKeyList will be write in search_params.params
+	// after 2.5.1, allow user to write all this key in search_params
+	// SearchParams in planpb.QueryInfo is the params set passed to segcore
+	// so if you want to use the params in segcore/knowhere, remember to add the new params into ParamsKeyList after 2.5.1
```

Comment:
Need add the index parameter here? This doesn't seem reasonable. It will introduce new compatibility logic. Can we just write all of params without adding a whitelist?